sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py

```diff
@@ -2596,3 +2598,25 @@ def my_pipeline() -> Output(type="uri_folder", description="new description", mo
             'description': 'new description', 'job_output_type': 'uri_folder', 'mode': 'Upload'
         }}
         assert pipeline_job._to_rest_object().as_dict()["properties"]["outputs"] == expected_outputs
+
+    def test_dsl_pipeline_run_settings(self) -> None:
+        hello_world_component_yaml = "./tests/test_configs/components/helloworld_component.yml"
+        hello_world_component_func = load_component(source=hello_world_component_yaml)
+
+        @dsl.pipeline()
+        def my_pipeline() -> Output(type="uri_folder", description="new description", mode="upload"):
+            node = hello_world_component_func(component_in_path=Input(path="path/on/ds"), component_in_number=10)
+            return {"output": node.outputs.component_out_path}
+
+        pipeline_job: PipelineJob = my_pipeline()
+        pipeline_job.settings.default_compute = "cpu-cluster"
+        pipeline_job.settings.continue_on_step_failure = True
```

Comment:
This feature deserves a changelog update

Reply:
Updated with: 
Fixed an issue that attributes with a value of `False` in `PipelineJobSettings` are not respected